table/server/underdb/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java

```diff
@@ -162,8 +162,12 @@ private PathTranslator mountAlluxioPaths(Table table, List<Partition> partitions
 
       for (Partition part : partitions) {
         AlluxioURI partitionUri;
-        if (part.getSd() != null && part.getSd().getLocation() != null
-            && ufsUri.isAncestorOf(partitionUri = new AlluxioURI(part.getSd().getLocation()))) {
+        if (part.getSd() != null && part.getSd().getLocation() != null) {
+          partitionUri = new AlluxioURI(part.getSd().getLocation());
+          if (!mConfiguration.getBoolean(Property.ALLOW_DIFF_PART_LOC_PREFIX)
+              && !ufsUri.isAncestorOf(partitionUri)) {
```

Comment:
Can this be a general property, since the glue integration also has a similar check: https://github.com/Alluxio/alluxio/blob/master/table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java#L263